### ISSUE TITLE
Paragraph whitespace stripping in bard field

### ISSUE THREAD

**Issue by LorensoD** [statamic/cms#10847]
### Bug description

After upgrading to Statamic 5, I’ve encountered an issue where the Bard field is stripping whitespace in certain situations. Specifically, when using bold text or hyperlinks within a paragraph, the spaces before and after the formatted text are removed, causing words to run together.

Bard field before saving:
<img width="825" alt="image" src="https://github.com/user-attachments/assets/73221f9d-7eec-4dc6-a3c2-7525208303ac">

Html source before saving:
<img width="825" alt="image" src="https://github.com/user-attachments/assets/2b00aba9-2343-4d66-9671-4ece6bd64a90">

Bard field after saving:
<img width="825" alt="image" src="https://github.com/user-attachments/assets/9cb0cf90-2d31-4db3-8432-bc7d27891ed3">

Html source after saving:
<img width="825" alt="image" src="https://github.com/user-attachments/assets/5c56d7d1-0827-4ff3-854d-05a9883459c6">

### How to reproduce

1. Create or edit an entry with a Bard field.
2. Type a paragraph of text.
3. Within the paragraph, select a word or phrase in the middle and apply bold formatting or insert a link.
4. Save the entry.
5. View the published content or revisit the editor.

### Environment

```yaml
Environment
Application Name: Statamic
Laravel Version: 11.21.0
PHP Version: 8.3.11
Composer Version: 2.7.7
Environment: local
Debug Mode: ENABLED
URL: --
Maintenance Mode: OFF
Timezone: UTC
Locale: en

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: file
Database: mysql
Logs: stack / single
Mail: smtp
Queue: redis
Session: redis

Sentry
Enabled: MISSING DSN
Environment: local
Laravel SDK Version: 4.8.0
PHP SDK Version: 4.9.0
Release: NOT SET
Sample Rate Errors: 100%
Sample Rate Performance Monitoring: NOT SET
Sample Rate Profiling: NOT SET
Send Default PII: DISABLED

Statamic
Addons: 2
Sites: 4 (Dutch, English, German, French)
Stache Watcher: Enabled
Static Caching: full
Version: 5.26.0 PRO

Statamic Addons
webographen/statamic-widget-continue-editing: 1.3.2
withcandour/aardvark-seo: 5.0.0
```

### Installation

Fresh statamic/statamic site via CLI

### Additional details

I use sets in my bard field.
The bard field does not Save as HTML.

**Comment by duncanmcclean** [comment]
Is this happening on the frontend too, or just when you view the source HTML in the Bard editor?

**Comment by LorensoD** [comment]
> Is this happening on the frontend too, or just when you view the source HTML in the Bard editor?

I'm using the Rest API. And there the rendered html in the json response is also missing whitespace.

**Comment by LorensoD** [comment]
After the V5 upgrade when I save a page with Bard fields, without making any content changes, it results in the whitespace being stripped. I get this change in the markdown/yaml content file of the page:

<img width="702" alt="image" src="https://github.com/user-attachments/assets/ff986d74-3916-4d6a-bf3e-90298bf574a6">

<img width="702" alt="image" src="https://github.com/user-attachments/assets/72a7aadd-b65a-4421-b5b9-2daaad3ea5ce">

**Comment by jasonvarga** [comment]
Please provide a repo with this problem reproduced.

**Comment by philippe-tellier** [comment]
Fixed by #11127
